dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -917,23 +939,76 @@ static void constructPartname(const char *filename,unsigned n, StringBuffer &pn,
 static bool parseFileName(const char *name,StringBuffer &mname,unsigned &num,unsigned &max,bool &replicate)
 {
     // takes filename and creates mask filename with $P$ extension
+    const char *filename = name;
     StringBuffer nonrepdir;
-    replicate = setReplicateDir(name,nonrepdir,false);
-    if (replicate) 
-        name = nonrepdir.str();
+    if (!isContainerized())
+    {
+        // Replicate dir is not supported in containerized environment
+        replicate = setReplicateDir(name,nonrepdir,false);
+        if (replicate)
+            name = nonrepdir.str();
+    }
+
+    Owned<IPropertyTreeIterator> planesIter = getPlanesIterator("data", nullptr);
+    ForEach(*planesIter)
+    {
+        const IPropertyTree &plane = planesIter->query();
+        const char *prefix = plane.queryProp("@prefix");
+        size_t prefixLen = strlen(prefix);
+        if (startsWith(name, prefix) && name[prefixLen] == PATHSEPCHAR)
+        {
+            mname.ensureCapacity(strlen(name));
+            mname.append(prefix).append(PATHSEPCHAR);
+            name += prefixLen + 1;
+            if (plane.getPropInt("@numDevices") > 1)
+            {
+                if (*name&&*name!='d')
+                    throw makeStringExceptionV(-1, "In storage plane definition numDevices>1, but no stripe sub-directory found in file %s", filename);
+                name++;
+                while (*name&&isdigit(*name))
+                    name++;
+                mname.append("d$P$");
+            }
+            break;
+        }
+    }
+
+    if (mname.isEmpty())
+        throw makeStringExceptionV(-1, "Could not find matching prefix in plane definition for file %s", filename);
```

Comment:
can the caller handle exceptions? I think not, or it looks like it will cause it to bail out of too much just because of 1 malformed physical filename.

Also, this function can now either throw exception or return false - which begs question how should they be considered differently? I think throwing an exception is fine if consistent.

I would remove the boolean result, throw an exception instead of the return false, and have the caller catch, log, skip further processing of this part, and continue

Reply:
NB: The alternative would be to CLogicalNameEntry create a dirpartmask that encapsulated an abstract stripe # and dirperpart, e.g. using $S for stripe.
e.g. so it filled dirpartmask with something like "/var/lib/HPCCSystems/hpcc-data/d$S$/mydata/sub/$P$/myfile._$P$_of_$N$" , and for parseFilename to do the same..

That would be a better approach I suspect, but requires more changes to processFile.

@jackdelv - happy to discuss further.

Reply:
I removed the boolean returns, and made exception throwing more consistent. The caller will now catch any error from parseFileName and log it with the manager.

I think it would fairly easy to build up a dirpartmask and return that alongside mname. Should that be done here, or in a follow up PR?

Reply:
let's leave for a subsequent PR. Can you open a separate JIRA to follow?